src/main/java/com/gsheet/student/service/GoogleAuthTokenProvider.java

```diff
@@ -0,0 +1,23 @@
+package com.gsheet.student.service;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.core.io.Resource;
+
+@RequiredArgsConstructor
+public class GoogleAuthTokenProvider implements AuthTokenProvider {
+    private final Resource credentialsResource;
+    private final String[] scopes;
+
+    @SneakyThrows
+    public String getAccessToken() {
```

Comment:
It is not the best logic to receive credentials. It is better to use Singleton pattern to create them, and each time you return them, check if they expired or not. It will works faster.